### PR TITLE
[5816] Switch to using zxcvbn for password strength management

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5816

### CODE REVIEW

File: ui/app/src/components/CreateUserDialog/CreateUserDialogContainer.tsx [hunk 1]
```diff
@@ -51,6 +51,7 @@ import {
 } from '../UserManagement/utils';
 import useUpdateRefs from '../../hooks/useUpdateRefs';
 import { showSystemNotification } from '../../state/actions/system';
+import zxcvbn from 'zxcvbn';
```
Comment (rart) [inline ui/app/src/components/CreateUserDialog/CreateUserDialogContainer.tsx]: This is a large lib which will unnecessarily grow the bundle. We need a async, on-demand mechanism to load it.

File: ui/app/src/components/CreateUserDialog/CreateUserDialogContainer.tsx [hunk 2]
```diff
@@ -214,7 +222,8 @@ export function CreateUserDialogContainer(props: CreateUserDialogContainerProps)
   };
 
   const isInvalidPassword = (password) => {
-    return !validPassword && password !== '';
+    const pw = zxcvbn(password);
+    return pw.score < passwordRequirementsMinComplexity && password !== '';
```
Comment (rart) [inline ui/app/src/components/CreateUserDialog/CreateUserDialogContainer.tsx]: Check blank before running the password complexity fn

File: ui/app/src/components/ResetPasswordDialog/ResetPasswordDialog.tsx [hunk 3]
```diff
@@ -36,6 +36,7 @@ interface ResetPasswordDialogProps {
   onClose(): void;
   user: User;
   passwordRequirementsRegex: string;
```
Comment (rart) [inline ui/app/src/components/ResetPasswordDialog/ResetPasswordDialog.tsx]: Why are we keeping this in some places?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/ResetPasswordDialog/ResetPasswordDialog.tsx (hunk 3 above)
Comment (jvega190) [reply ui/app/src/components/ResetPasswordDialog/ResetPasswordDialog.tsx]: They were not removed yet. Done

File: ui/app/src/components/CreateUserDialog/CreateUserDialogContainer.tsx (hunk 1 above)
Comment (jvega190) [reply ui/app/src/components/CreateUserDialog/CreateUserDialogContainer.tsx]: still pending

File: ui/app/src/components/AuthBoundary/AuthBoundary.tsx [hunk 4]
```diff
@@ -39,11 +39,7 @@ export function AuthBoundary(props) {
     return (
       <I18nProvider>
         <CrafterThemeProvider>
-          <Login
-            passwordRequirementsRegex="'^(?=(?<hasNumbers>.*[0-9]))(?=(?<hasLowercase>.*[a-z]))(?=(?<hasUppercase>.*[A-Z]))(?=(?<hasSpecialChars>.*[~|!`,;\/@#$%^&+=]))(?<minLength>.{8,})$'"
-            xsrfToken={getRequestForgeryToken()}
-            xsrfParamName="_csrf"
-          />
+          <Login passwordRequirementsMinComplexity={4} xsrfToken={getRequestForgeryToken()} xsrfParamName="_csrf" />
```
Comment (rart) [inline ui/app/src/components/AuthBoundary/AuthBoundary.tsx:42]: Hardcoded min complexity prop?

**Review by rart — COMMENTED** [review]
Restore/leave the PasswordRequirementsDisplay component

File: ui/app/src/components/PasswordStrengthDisplay/PasswordStrengthDisplay.tsx [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+    onChangeTimeoutRef.current = setTimeout(() => {
+      import('zxcvbn').then(({ default: zxcvbn }) => {
+        setPassword(zxcvbn(value));
+      });
+    }, 200);
+  }, [value]);
+
+  useEffect(() => {
+    onValidStateChanged(isBlank(value) ? null : password?.score >= passwordRequirementsMinComplexity);
+  }, [onValidStateChanged, value, password, passwordRequirementsMinComplexity]);
```
Comment (rart) [inline ui/app/src/components/PasswordStrengthDisplay/PasswordStrengthDisplay.tsx]: Couldn't we eliminate this effect and do it all when the `zxcvbn()` is ran?

**Review by rart — APPROVED** [review]

**Pull request merged**